ambry-store/src/main/java/com.github.ambry.store/CompactionLog.java

```diff
@@ -336,21 +350,23 @@ Phase getPhase() {
 
         compactionDetails (see CompactionDetails#toBytes())
         copyStartTime
-        switchStartTime
+        commitStartTime
         cleanupStartTime
         cycleEndTime
         safeToken (see StoreFindToken#toBytes())
        */
       byte[] compactionDetailsBytes = compactionDetails.toBytes();
-      byte[] safeTokenBytes = safeToken.toBytes();
-      int size = compactionDetailsBytes.length + 4 * TIMESTAMP_SIZE + safeTokenBytes.length;
+      byte[] safeTokenBytes = safeToken != null ? safeToken.toBytes() : ZERO_LENGTH_ARRAY;
```

Comment:
Or if we can rely on callers, that compactionLog.setSafeToken() will always be called with UnInitialized token, we can set this to Unintialized. But I am ok either ways